texastribune/data-visuals-create
# Add last build time

## [4.0.0]
- Possibly a major version bump because we need the updatedDate for future versions of the graphics plugin? What do you think @ashley-hebler?

### Changed
- `templates/__common__/config/scripts/build.js` - add script to update the `lastBuildTime` in `build.js`
- `templates/feature/project.config.js` - add `lastBuildTime` field to feature config, change `slug` to `projectName`, add comments to clarify which properties should and should not be changed
- `templates/graphic/project.config.js` - add `lastBuildTime` field to graphic config, change `slug` to `projectName`, add comments to clarify which properties should and should not be changed
- `bin/data-visuals-create`, `README.md`, `templates/__common__/_package.json`, `templates/__common__/utils/deployment/update-log-sheet.js` - change `slug` to `projectName` to avoid confusion with the slug in our project URL and what we add to the CMS

### Added
- `templates/__common__/config/tasks/last-build-time.js` - add script to parse the config file, update the `lastBuildTime` and write the file back to the filepath

**To test:**
1) Check out `add-updated-date` and run `your/path/to/data-visuals-create/bin/data-visuals-create graphic my-test-graphic`
2) Run `npm run build`
3) Check that the `updatedDate` field in `project.config.js` was changed to the current timestamp
4) ^ Repeat the above steps for a feature

@ashley-hebler would love your thoughts on if I'm handling errors correctly in the `updated-date.js` script!
